src/NetworkAlias.php

```diff
@@ -399,6 +399,7 @@ public static function showForFQDN(CommonGLPI $item, $withtemplate)
 
         $alias   = new self();
         $address = new NetworkName();
+        /** @var CommonDBTM $item */
```

Comment:
I don't know if manually specifying the type is the correct solution.

PHPstan is telling us that something is wrong because the method accept a `CommonGLPI` parameter but we use it as a `CommonDBTM` object.

IMO using `@var` does not fix the underlying issue but just hides it (which defeat the point of static analysis).

It would be cleaner to execute a `instance of CommonDBTM` check, with an early return or exception if the item does not pass the check.

Since `showForFQDN` is a method specific to this file, the parameter type of `$item` could even be set directly to `CommonDBTM`, with the `instanceOf` check being done before the method is called in `displayTabContentForItem`.

Reply:
I'm not going to spend any extra time speaking of that kind of dumb change; I'll just revert. Adding type or changing anything in the existing code will just be a nightmare - the point was to pass phpstan to another level so we can get bugs fixed.

Many times, it's not possible to replace CommonGLPI with CommonDBTM, because in some implementation, we really have one or the other.

I give up; that's far out of the scope - we'll see later if we do something.